File: src/main/java/com/sport/service/annotations/PhoneNumberValid.java
package com.sport.service.annotations;

import com.sport.service.validators.PhoneNumberValidator;
import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.*;

@Documented
@Constraint(validatedBy = PhoneNumberValidator.class)
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface PhoneNumberValid {

    String message() default "Invalid phone number format. Example: +7XXXXXXXXXX";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/com/sport/service/validation/joint_training/PhoneNumberValid.kt
package com.sport.service.validation.joint_training

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [PhoneNumberValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class PhoneNumberValid(
    val message: String = "Invalid phone number format. Example: +7XXXXXXXXXX",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)